### PR TITLE
hourly pub rank in statistics page

### PR DESCRIPTION
**Pull request by DoraALin**
1. bring hourly pub size back to statistics page
2. cluster info testcase
Signed-off-by: 元守 <lulin@youzan.com>

### CODE REVIEW

**Review by absolute8511 — COMMENTED** [review]
Pls improve the performance by aggregating the history stats api

**Comment by DoraALin** [comment]
/message/hisrotystats API added in nsqd to read last hourly pubsize for all writable topics.
@absolute8511  pls help review.

**Comment by absolute8511** [comment]
pls add test case for  the new  doMessageHistoryStats and bench it for the necessary  for goroutines.

File: nsqdserver/http.go [hunk 1]
```diff
@@ -646,26 +648,77 @@ func (s *httpServer) doMessageHistoryStats(w http.ResponseWriter, req *http.Requ
...
-		cur++
+	if topicName == "" && topicPartStr == "" {
+
+		var lock sync.Mutex
+		var wg sync.WaitGroup
+		topicStats := s.ctx.getStats()
+		var topicHourlyPubStatsList []*clusterinfo.NodeHourlyPubsize
+		for _, topicStat := range topicStats {
+			wg.Add(1)
+			go func(topicStat nsqd.TopicStats) {
```
Comment (absolute8511) [inline nsqdserver/http.go]: Do we really need goroutines here? Maybe benchmark it.

Comment (DoraALin) [reply nsqdserver/http.go]: 1. benchmark added for sync node message history invoke;
2. leader filter in NSQd stats API;
@absolute8511 for your review

**Comment by DoraALin** [comment]
@absolute8511 for your review
1. filter filter in nsqd getStats to filter out node which is write disabled;
2. testcase associated;

File: nsqdserver/http_test.go [hunk 2]
```diff
@@ -597,3 +598,185 @@ func TestHTTPerrors(t *testing.T) {
...
+	json.Unmarshal(body, &statFiltered)
+	t.Logf("topic len: %d", len(statFiltered.TopicStats))
+	test.Assert(t, len(statFiltered.TopicStats) == 1, "topic number is not empty.")
+}
+
+func BenchmarkFetchNodeHourlyPubsize(b *testing.B) {
+	b.StopTimer()
+	lgr := newTestLogger(b)
+
+	nsqlookupdOpts := nsqlookupd.NewOptions()
```
Comment (absolute8511) [inline nsqdserver/http_test.go]: I think benchmark the doMessageHistoryStats interface should not use the nsqlookupd. Pls make it simple.

Comment (DoraALin) [reply nsqdserver/http_test.go]: @absolute8511, done. There is only nsqd instance in benchmark

**Review by absolute8511 — APPROVED** [review]
LGTM

**Pull request merged**